linkerd/app/outbound/src/http/detect.rs

```diff
@@ -54,48 +46,23 @@ impl<T> Outbound<T> {
             .push(http::NewServeHttp::layer(h2_settings, rt.drain.clone()))
             .push_map_target(http::Logical::from)
             .push(svc::UnwrapOr::layer(
-                // When an HTTP version cannot be detected, we fallback to a logical
-                // TCP stack. This service needs to be buffered so that it can be
-                // cached and cloned per connection.
-                svc::stack(tcp.clone())
-                    .push(profiles::split::layer())
-                    .push_switch(ShouldResolve, tcp_forward.clone())
-                    .push_on_response(
-                        svc::layers()
-                            .push_map_target(io::EitherIo::Right)
-                            .push(rt.metrics.stack.layer(stack_labels("tcp", "logical")))
-                            .push(svc::layer::mk(svc::SpawnReady::new))
-                            .push(svc::FailFast::layer("TCP Logical", dispatch_timeout))
-                            .push_spawn_buffer(buffer_capacity),
-                    )
-                    .instrument(|_: &_| debug_span!("tcp"))
-                    .check_new_service::<tcp::Logical, _>()
+                tcp.clone()
+                    .push_on_response(svc::layers().push_map_target(io::EitherIo::Right))
                     .into_inner(),
             ))
-            .push_cache(cache_max_idle_age)
             .push(detect::NewDetectService::layer(
                 detect_protocol_timeout,
                 http::DetectHttp::default(),
             ))
-            .check_new_service::<tcp::Logical, _>()
+            // When the profile marks the target as opaque, we skip HTTP
+            // detection and just use the TCP logical stack directly. Unlike the
+            // above case, this stack need not be buffered, since `fn cache`
+            // applies its own buffer on the returned service.
```

Comment:
is this comment still accurate here? we no longer push a buffer in the `UnwrapOr` above either...
```suggestion
            // detection and just use the TCP logical stack directly.
```